src/project/loadWorkspace.js

```diff
@@ -82,13 +75,27 @@ function readPnpmWorkspaceYamlIfExists(dir) {
  * @returns {import('./project-types.js').PartialWorkspace}
  */
 export function loadWorkspace(dir) {
-  const packageJson = readPackageJsonIfExists(dir)
+  let packageJson
+  try {
+    packageJson = readPackageJsonIfExists(dir)
+  } catch (err) {
+    throw logger.fail(`Failed reading package.json in '${dir}'`, {
+      detail: err instanceof Error ? err.message : undefined,
+    })
+  }
   if (!packageJson) {
-    throw new Error(`Could not find package.json in ${dir}`)
+    throw new Error(`Could not find package.json in '${dir}'`)
+  }
+  let pnpmWorkspaceYaml
+  try {
+    pnpmWorkspaceYaml = readPnpmWorkspaceYamlIfExists(dir)
+  } catch (err) {
+    throw logger.fail(`Failed reading pnpm-workspace.yaml in '${dir}'`, {
+      detail: err instanceof Error ? err.message : undefined,
+    })
   }
-  const pnpmWorkspaceYaml = readPnpmWorkspaceYamlIfExists(dir)
   if (packageJson.workspaces && pnpmWorkspaceYaml) {
-    throw new Error(`Both pnpm-workspace.yaml and package.json workspaces are defined in ${dir}`)
+    throw new Error(`Both pnpm-workspace.yaml and package.json workspaces are defined in '${dir}'`)
```

Comment:
Do we want to throw error or switch to using `logger.fail` here and in the cases above?

Reply:
yeah, `logger.fail` should be used for all errors that we anticipate and provide an error message for.

Reply:
Right, I will adjust it and push the changes.